src/components/pages/UploadContacts/index.tsx

```diff
@@ -0,0 +1,259 @@
+                  target="_blank"
+                  rel="noopener noreferrer">
+                  Download our CSV template
+                </a>
+              </span>
+              <Image src={Template} className="mb-5" />
+              <Docdrop
+                uploadFile={uploadCsv}
+                uploadedFile={uploadedCsv.file}
+                acceptedFormat="text/csv/*"
+                acceptedFormatLabel="CSV"
+              />
+              <div className="ml-auto">
+                <FunnelButton
+                  cta="Next"
+                  onNext={handleNextClick}
+                  enabled={uploadedCsv.file !== null}
+                />
+              </div>
+            </div>
+          )}
+
+          {uploadStep === 1 && (
+            <div className="upload-file-step-container">
+              <span>
+                Let's make sure that the columns in your uploaded csv map to the
+                appropriate fields.
+              </span>
+              <div className="w-100 shadow-sm p-3 mt-3">
+                <FieldMappingTable
+                  headers={uploadedCsv.header}
+                  sample={uploadedCsv.data[0]}
+                  mapping={mapping}
+                  setMapping={setMapping}
+                />
+              </div>
+              <FunnelButton
+                cta="Next"
+                onNext={handleNextClick}
+                onBack={handleBackClick}
+                enabled={Object.values(mapping).every(
+                  (fieldMap) => (fieldMap as FieldMap).index >= 0,
+                )}
+              />
+            </div>
+          )}
+
+          {uploadStep === 2 && (
+            <div className="upload-file-step-container">
+              <span>
+                Assign existing tags or create new ones for your contacts.
+              </span>
+              <TagSelector
+                tags={tags}
+                selectedTags={selectedTags}
+                addTag={addUploadTag}
+                removeTag={removeUploadTag}
+                showInputField={true}
+                createTag={createTag}
+              />
+              <FunnelButton
+                cta="Create contacts"
+                onNext={handleSubmission}
+                onBack={handleBackClick}
+                enabled={uploadedCsv !== null}
+              />
+            </div>
+          )}
+          {uploadStep === 3 && (
+            <div className="upload-file-step-container">
+              <span className="black-500 p3 mb-3 font-weight-semibold">
+                We're adding your contacts!
+              </span>
+              <Clock size={60} color="#C3DBF5" />
+              <span className="mt-3 text-center">
+                Depending on the number of contacts you've uploaded, this may
+                take several minutes. You can move on to your next tasks in the
+                mean time.
```

Comment:
i'm kinda too lazy to dig into this, but how would the user be notified if they moved on and then the uploaded succeeded / failed? Will an alert pop up on screen?

Reply:
Don't have that functionality yet! but my plan is to add some variables to the redux store to let me know once a big request like this is done. I'll add it as a todo